### PR TITLE
Handled all basic rules

### PR DESCRIPTION
**Pull request by locmai**
Ticket: https://github.com/sieuvb/o-an-quan/issues/2

Demo returned result: https://gist.github.com/locmai/cf102363369620a609ffd7a9c116e93f

### CODE REVIEW

File: packages/backend/src/services/GameService.ts [hunk 1]
```diff
@@ -95,6 +112,212 @@ export class GameService {
...
+    game = gameRepository.switchTurn(roomId);
+
+    // Check if both Big square is blank.
+    if (
+      this.squareIsBlank(game.gameState?.squares[0]) &&
+      this.squareIsBlank(game.gameState?.squares[11])
+    ) {
+      game = gameRepository.updateRoomStatus(roomId, RoomStatus.FINISHED);
+      return game;
+    }
```
Comment (helluuuuu) [inline packages/backend/src/services/GameService.ts]: Are big square's indexes 5 and 11 instead?

File: packages/backend/src/services/GameService.ts [hunk 2]
```diff
@@ -95,6 +112,212 @@ export class GameService {
...
+        calculatedSquares[selectedSquareIndex].smallStoneNum;
+
+      // Pick up all the stones in the selected square
+      calculatedSquares[selectedSquareIndex].smallStoneNum = 0;
+      gameStep.steps.push({
+        action: StepAction.MOVE,
+        squareId: selectedSquareIndex,
+        smallStoneNum: calculatedSquares[selectedSquareIndex].smallStoneNum,
+        bigStoneNum: calculatedSquares[selectedSquareIndex].bigStoneNum,
+      });
```
Comment (helluuuuu) [inline packages/backend/src/services/GameService.ts]: Is this redundant?

File: packages/backend/src/repositories/GameRepository.ts [hunk 3]
```diff
@@ -67,6 +72,57 @@ export class GameRepository {
...
+  takeSmallStoneFromPlayer = (
+    roomId: string,
+    numOfSmallStonesTaken: number,
+  ) => {
+    const roomInfo = this.roomsData[roomId];
+    const currentTurn = roomInfo.gameState?.currentTurn;
+    const players = roomInfo.gameState?.players;
+    let smallStoneNum = roomInfo.gameState.players[currentTurn].playerGameInfo.smallStoneNum
+    smallStoneNum = smallStoneNum - numOfSmallStonesTaken;
+    return roomInfo;
```
Comment (helluuuuu) [inline packages/backend/src/repositories/GameRepository.ts:111]: Is it really working? 🤪

File: packages/backend/src/services/GameService.ts (hunk 1 above)
Comment (locmai) [reply packages/backend/src/services/GameService.ts]: oh it is. Nice catch

File: packages/backend/src/services/GameService.ts (hunk 2 above)
Comment (locmai) [reply packages/backend/src/services/GameService.ts]: changed to 0, shorter.

**Pull request merged**